dspace-spring-rest/src/test/java/org/dspace/app/opensearch/OpenSearchControllerTest.java

```diff
@@ -0,0 +1,144 @@
+import static org.hamcrest.Matchers.not;
+*/
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.xpath;
+
+/*
+import org.dspace.app.rest.OpenSearchController;
+import org.dspace.services.ConfigurationService;
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.GroupBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.matcher.BrowseEntryResourceMatcher;
+import org.dspace.app.rest.matcher.BrowseIndexMatcher;
+import org.dspace.app.rest.matcher.ItemMatcher;
+import org.dspace.app.util.service.OpenSearchService;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.eperson.Group;
+*/
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+
+import org.dspace.services.ConfigurationService;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Integration test to test the /opensearch endpoint
+ * (Class has to start or end with IT to be picked up by the failsafe plugin)
+ *
+ * @author Oliver Goldschmidt (o dot goldschmidt at tuhh dot de)
+ */
+public class OpenSearchControllerTest extends AbstractControllerIntegrationTest {
+
+    // configuration is taken from dspace-api/src/test/data/dspaceFolder/config/local.cfg
+    private ConfigurationService configurationService;
+
+    @Before
+    public void init() throws Exception {
+        // override the configuration settings here if other settings are required for test
+        // configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        // configurationService.setProperty("websvc.opensearch.enable", true);
+        /*
+        System.out.println("Testing OpenSearch");
+        MockHttpServletResponse resp2 = getClient().perform(get("/opensearch/search")
+            .param("query", "cats"))
+            .andReturn().getResponse();
+        System.out.println("Response from Test 2: "+resp2.getContentAsString());
+        */
+    }
+
+    @Test
+    public void searchAtomTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "cats"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/atom+xml;charset=UTF-8"
+                   .andExpect(content().contentType("application/atom+xml;charset=UTF-8"))
+        ;
+    }
+
+    /* HTML is an open issue in OpenSearch, so skip this test at the moment
```

Comment:
This is a good explanation as to why we have this test commented out. Thanks!

Reply:
Glad to hear, thanks :-)